mkosi/burn.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import sys
+
+from mkosi.config import (
+    MkosiArgs,
+    MkosiConfig,
+    OutputFormat,
+)
+from mkosi.log import complete_step, die
+from mkosi.run import run
+
+
+def run_burn(args: MkosiArgs, config: MkosiConfig) -> None:
+    if config.output_format not in (OutputFormat.disk, OutputFormat.esp):
+        die(f"{config.output_format} images cannot be dd'ed to disk")
+
+    fname = config.output_dir_or_cwd() / config.output
+
+    if (len(args.cmdline) != 1):
+            die("Expected device argument.");
```

Comment:
No unneeded parentheses around conditions and the indentation is a bit too far (8 instead of 4).
```suggestion
    if len(args.cmdline) != 1:
        die("Expected device argument.");
```